src/com/dotcms/rest/api/v1/sites/ruleengine/rules/RuleResource.java

```diff
@@ -237,7 +233,13 @@ private User getUser(@Context HttpServletRequest request) {
 
     private RestRule getRuleInternal(String ruleId, User user) {
         Rule input = getRule(ruleId, user);
-        return ruleTransform.appToRest(input);
+        RestRule restRule = ruleTransform.appToRest(input, user);
+
+        if(restRule == null) {
```

Comment:
the transform should never ever return null. if something changed we need to unchange it. 

#getRule(...) should throw a NotFoundException if the ruleId isn't present in the database. The transform.appToRest should throw any errors that prevent the method from returning a valid rule.